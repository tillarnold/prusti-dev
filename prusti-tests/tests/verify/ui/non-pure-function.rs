// ignore-test
#![allow(dead_code)]
use prusti_contracts::*;

mod foo {
    pub fn get_false() -> bool {
        false
    }
}

fn get_true() -> bool {
    true
}

#[pure]
fn pure_get_true() -> bool {
    true
}

#[requires(get_true())]
fn foo() {}

#[requires(pure_get_true() && !foo::get_false())]
fn main() {}
