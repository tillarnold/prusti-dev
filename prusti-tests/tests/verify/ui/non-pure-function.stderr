error: [Prusti: invalid specification] use of impure function "get_true" in pure code is not allowed
  --> $DIR/non-pure-function.rs:19:12
   |
19 | #[requires(get_true())]
   |            ^^^^^^^^^^

error: [Prusti: invalid specification] use of impure function "foo::get_false" in pure code is not allowed
  --> $DIR/non-pure-function.rs:22:32
   |
22 | #[requires(pure_get_true() && !foo::get_false())]
   |                                ^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

